packages/scene-composer/src/components/panels/scene-components/DataOverlayComponentEditor.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { act, render } from '@testing-library/react';
+import wrapper from '@awsui/components-react/test-utils/dom';
+
+import { KnownComponentType } from '../../../interfaces';
+import { IDataOverlayComponentInternal, ISceneNodeInternal, useStore } from '../../../store';
+import { Component } from '../../../models/SceneModels';
+import { mockProvider } from '../../../../tests/components/panels/scene-components/MockComponents';
+
+import { DataOverlayComponentEditor } from './DataOverlayComponentEditor';
+
+jest.mock('@awsui/components-react', () => ({
+  ...jest.requireActual('@awsui/components-react'),
+}));
+
+jest.mock('./data-overlay/DataBindingMapEditor', () => {
+  const originalModule = jest.requireActual('./data-overlay/DataBindingMapEditor');
+  return {
+    ...originalModule,
+    DataBindingMapEditor: (...props: any[]) => {
+      return <div data-testid='DataBindingMapEditor'>{JSON.stringify(props)}</div>;
+    },
+  };
+});
+
+describe('DataBindingMapEditor', () => {
+  const component: IDataOverlayComponentInternal = {
+    ref: 'comp-ref',
+    type: KnownComponentType.DataOverlay,
+    subType: Component.DataOverlaySubType.TextAnnotation,
+    dataRows: [
+      {
+        rowType: Component.DataOverlayRowType.Markdown,
+        content: 'markdown-content',
+      },
+    ],
+    valueDataBindings: [],
+  };
+  const node = {
+    ref: 'node-ref',
+  } as ISceneNodeInternal;
+  const updateComponentInternalMock = jest.fn();
+
+  const baseState = {
+    updateComponentInternal: updateComponentInternalMock,
+    getEditorConfig: jest.fn().mockReturnValue({ valueDataBindingProvider: mockProvider }),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
```

Comment:
This shouldn't need to be explicitly called in most circumstances, it's a code smell that results from sharing objects between tests.

Each test should setup it's own behaviors that we care about validating (things we don't care can be mocked at the root, but we won't be making assertions on them). For example, a much cleaner way to test the update behavior in the component below is to isolate the update callback.

You can do this by mocking `useCallback` and grabbing the callback passed to it, and just executing it directly. You don't need to traverse the DOM at all or care about the DOM structure to validate it.

Then you add a second test that validates the DOM structure via a snpahsot, and we've effectively covered this components full set of responsibilities.